# Introduction

In this notebook, we will do a comprehensive analysis of the Android app market by comparing thousands of apps in the Google Play store.

# About the Dataset of Google Play Store Apps & Reviews

**Data Source:** <br>
App and review data was scraped from the Google Play Store by Lavanya Gupta in 2018. Original files listed [here](
https://www.kaggle.com/lava18/google-play-store-apps).

# Import Statements

In [41]:
import pandas as pd
import plotly.express as px
import numpy as np

# Notebook Presentation

In [42]:
# Show numeric output in decimal format e.g., 2.15
pd.options.display.float_format = '{:,.2f}'.format

# Read the Dataset

In [43]:
df_apps = pd.read_csv('apps.csv')

# Data Cleaning

**Challenge**: How many rows and columns does `df_apps` have? What are the column names? Look at a random sample of 5 different rows with [.sample()](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sample.html).

In [44]:
df_goog_play = pd.read_csv("apps.csv")

In [45]:
print(df_goog_play.shape) # rows and columns
# print vs display outputs - no visible differnce here but I think that display is good for displaying tables, 
# print is not
display(df_goog_play.columns) # Names of columns
print(df_goog_play.columns)

(10841, 12)


Index(['App', 'Category', 'Rating', 'Reviews', 'Size_MBs', 'Installs', 'Type',
       'Price', 'Content_Rating', 'Genres', 'Last_Updated', 'Android_Ver'],
      dtype='object')

Index(['App', 'Category', 'Rating', 'Reviews', 'Size_MBs', 'Installs', 'Type',
       'Price', 'Content_Rating', 'Genres', 'Last_Updated', 'Android_Ver'],
      dtype='object')


In [46]:
# Demonstration of difference betweem display and print in Jupyter Notebook
display(df_goog_play.sample(5)) # prints nice table
print(df_goog_play.sample(5)) # Not visually appealing or easy to read

App            Category  Rating  Reviews  \
2513   BGKontakti Bayern BG Kontakti              SOCIAL     NaN       32   
10421                       LinkedIn              SOCIAL    4.20  1225339   
204                       AQ Dentals  HEALTH_AND_FITNESS     NaN        0   
3708    DB Welt - Die Zeitung der DB  NEWS_AND_MAGAZINES     NaN      145   
985                      DW Security            BUSINESS    5.00        6   

       Size_MBs     Installs  Type Price Content_Rating            Genres  \
2513       7.80        1,000  Free     0       Everyone            Social   
10421      5.30  100,000,000  Free     0       Everyone            Social   
204       12.00           10  Free     0       Everyone  Health & Fitness   
3708      16.00       10,000  Free     0       Everyone  News & Magazines   
985       15.00          100  Free     0       Everyone          Business   

            Last_Updated Android_Ver  
2513        May 25, 2016  3.0 and up  
10421     August 2, 2018  5.0 and up  
204    December 22, 2017  4.1 and up  
3708   February 26, 2016  4.0 and up  
985        July 25, 2018  4.1 and up

                                                     App            Category  \
3366                             SAMSUNG RETAILMODE 2018  LIBRARIES_AND_DEMO   
7482                          Game of Thrones: Conquest™                GAME   
10247  DU Recorder – Screen Recorder, Video Editor, Live       VIDEO_PLAYERS   
8756                       NARUTO X BORUTO NINJA VOLTAGE                GAME   
4303                                          Eat'n Park           LIFESTYLE   

       Rating  Reviews  Size_MBs    Installs  Type Price Content_Rating  \
3366     4.30        6      5.50      10,000  Free     0       Everyone   
7482     4.00    90218     97.00   1,000,000  Free     0           Teen   
10247    4.80  2588730      9.70  50,000,000  Free     0       Everyone   
8756     4.40   267395     71.00   5,000,000  Free     0   Everyone 10+   
4303     3.50      292     20.00      50,000  Free     0       Everyone   

                        Genres       Last_Updated Android_Ver  
3366

### Drop Unused Columns

**Challenge**: Remove the columns called `Last_Updated` and `Android_Version` from the DataFrame. We will not use these columns. 

In [47]:
df_goog_play.drop(["Last_Updated", "Android_Ver"], axis=1)
df_goog_play # need to use inplace = True to truly remove rows
df_goog_play.drop(["Last_Updated", "Android_Ver"], axis=1, inplace=True)



In [48]:
df_goog_play

App         Category  Rating  \
0                           Ak Parti Yardım Toplama           SOCIAL     NaN   
1                        Ain Arabic Kids Alif Ba ta           FAMILY     NaN   
2      Popsicle Launcher for Android P 9.0 launcher  PERSONALIZATION     NaN   
3                         Command & Conquer: Rivals           FAMILY     NaN   
4                                        CX Network         BUSINESS     NaN   
...                                             ...              ...     ...   
10836                                Subway Surfers             GAME    4.50   
10837                                Subway Surfers             GAME    4.50   
10838                                Subway Surfers             GAME    4.50   
10839                                Subway Surfers             GAME    4.50   
10840                                Subway Surfers             GAME    4.50   

        Reviews  Size_MBs       Installs  Type   Price Content_Rating  \
0             0      8.70              0  Paid  $13.99           Teen   
1             0     33.00              0  Paid   $2.99       Everyone   
2             0      5.50              0  Paid   $1.49       Everyone   
3             0     19.00              0   NaN       0   Everyone 10+   
4             0     10.00              0  Free       0       Everyone   
...         ...       ...            ...   ...     ...            ...   
10836  27723193     76.00  1,000,000,000  Free       0   Everyone 10+   
10837  27724094     76.00  1,000,000,000  Free       0   Everyone 10+   
10838  27725352     76.00  1,000,000,000  Free       0   Everyone 10+   
10839  27725352     76.00  1,000,000,000  Free       0   Everyone 10+   
10840  27711703     76.00  1,000,000,000  Free       0   Everyone 10+   

                Genres  
0               Social  
1            Education  
2      Personalization  
3             Strategy  
4             Business  
...                ...  
10836           Arcade  
10837           Arcade  
10838           Arcade  
10839           Arcade  
10840           Arcade  

[10841 rows x 10 columns]

### Find and Remove NaN values in Ratings

**Challenge**: How may rows have a NaN value (not-a-number) in the Ratings column? Create DataFrame called `df_apps_clean` that does not include these rows. 

In [49]:
display(df_goog_play[df_goog_play["Rating"].isna()].sample(5))
df_goog_play[df_goog_play["Rating"].isna()].count()


App         Category  Rating  \
2234                                FO AIRBUS TLSE    COMMUNICATION     NaN   
787                                        BV Link         BUSINESS     NaN   
1122  Wallpapers New AK 47 Assault Rifle Guns Arms  PERSONALIZATION     NaN   
746                                     music (CG)           FAMILY     NaN   
2322   Bg Radios - Bulgarian radio stations online           FAMILY     NaN   

      Reviews  Size_MBs Installs  Type Price Content_Rating           Genres  
2234        4     16.00    1,000  Free     0       Everyone    Communication  
787        10      2.80      100  Free     0       Everyone         Business  
1122        0     16.00      100  Free     0           Teen  Personalization  
746         0     99.00       50  Free     0       Everyone    Entertainment  
2322        8     13.00    1,000  Free     0       Everyone    Entertainment

App               1474
Category          1474
Rating               0
Reviews           1474
Size_MBs          1474
Installs          1474
Type              1473
Price             1474
Content_Rating    1474
Genres            1474
dtype: int64

In [50]:
df_apps_clean = df_goog_play[df_goog_play["Rating"].isna()==False]

### Find and Remove Duplicates

**Challenge**: Are there any duplicates in data? Check for duplicates using the [.duplicated()](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.duplicated.html) function. How many entries can you find for the "Instagram" app? Use [.drop_duplicates()](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.drop_duplicates.html) to remove any duplicates from `df_apps_clean`. 


In [51]:
df_apps_clean.duplicated()
df_apps_clean[df_apps_clean["App"]=="Instagram"].count()
df_apps_clean.drop_duplicates(keep='first', inplace=True) # initial
df_apps_clean

C:\Users\btl\AppData\Local\Temp\ipykernel_19848\1856241806.py:3: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



App Category  Rating  \
21                                 KBA-EZ Health Guide  MEDICAL    5.00   
28                                            Ra Ga Ba     GAME    5.00   
47                                             Mu.F.O.     GAME    5.00   
82                                    Brick Breaker BR     GAME    5.00   
99     Anatomy & Physiology Vocabulary Exam Review App  MEDICAL    5.00   
...                                                ...      ...     ...   
10835                                   Subway Surfers     GAME    4.50   
10836                                   Subway Surfers     GAME    4.50   
10837                                   Subway Surfers     GAME    4.50   
10838                                   Subway Surfers     GAME    4.50   
10840                                   Subway Surfers     GAME    4.50   

        Reviews  Size_MBs       Installs  Type  Price Content_Rating   Genres  
21            4     25.00              1  Free      0       Everyone  Medical  
28            2     20.00              1  Paid  $1.49       Everyone   Arcade  
47            2     16.00              1  Paid  $0.99       Everyone   Arcade  
82            7     19.00              5  Free      0       Everyone   Arcade  
99            1      4.60              5  Free      0       Everyone  Medical  
...         ...       ...            ...   ...    ...            ...      ...  
10835  27722264     76.00  1,000,000,000  Free      0   Everyone 10+   Arcade  
10836  27723193     76.00  1,000,000,000  Free      0   Everyone 10+   Arcade  
10837  27724094     76.00  1,000,000,000  Free      0   Everyone 10+   Arcade  
10838  27725352     76.00  1,000,000,000  Free      0   Everyone 10+   Arcade  
10840  27711703     76.00  1,000,000,000  Free      0   Everyone 10+   Arcade  

[8891 rows x 10 columns]

In [52]:
display(df_apps_clean[df_apps_clean["App"]=="Instagram"])
# Still 3 copies of Instagram
df_apps_clean = df_apps_clean.drop_duplicates(keep='first',subset=["App"])
display(df_apps_clean[df_apps_clean["App"]=="Instagram"])
df_apps_clean.shape

App Category  Rating   Reviews  Size_MBs       Installs  Type  \
10806  Instagram   SOCIAL    4.50  66577313      5.30  1,000,000,000  Free   
10808  Instagram   SOCIAL    4.50  66577446      5.30  1,000,000,000  Free   
10810  Instagram   SOCIAL    4.50  66509917      5.30  1,000,000,000  Free   

      Price Content_Rating  Genres  
10806     0           Teen  Social  
10808     0           Teen  Social  
10810     0           Teen  Social

App Category  Rating   Reviews  Size_MBs       Installs  Type  \
10806  Instagram   SOCIAL    4.50  66577313      5.30  1,000,000,000  Free   

      Price Content_Rating  Genres  
10806     0           Teen  Social

(8197, 10)

# Find Highest Rated Apps

**Challenge**: Identify which apps are the highest rated. What problem might you encounter if you rely exclusively on ratings alone to determine the quality of an app?

In [53]:
df_apps_clean[df_apps_clean["Rating"] == df_apps_clean["Rating"].max()] # apps with highest rating

App   Category  Rating  \
21                                 KBA-EZ Health Guide    MEDICAL    5.00   
28                                            Ra Ga Ba       GAME    5.00   
47                                             Mu.F.O.       GAME    5.00   
82                                    Brick Breaker BR       GAME    5.00   
99     Anatomy & Physiology Vocabulary Exam Review App    MEDICAL    5.00   
...                                                ...        ...     ...   
2680                               Florida Wildflowers     FAMILY    5.00   
2750  Superheroes, Marvel, DC, Comics, TV, Movies News     COMICS    5.00   
3030           CL Keyboard - Myanmar Keyboard (No Ads)      TOOLS    5.00   
3115                                        Oración CX  LIFESTYLE    5.00   
4058                          Ek Bander Ne Kholi Dukan     FAMILY    5.00   

      Reviews  Size_MBs Installs  Type  Price Content_Rating         Genres  
21          4     25.00        1  Free      0       Everyone        Medical  
28          2     20.00        1  Paid  $1.49       Everyone         Arcade  
47          2     16.00        1  Paid  $0.99       Everyone         Arcade  
82          7     19.00        5  Free      0       Everyone         Arcade  
99          1      4.60        5  Free      0       Everyone        Medical  
...       ...       ...      ...   ...    ...            ...            ...  
2680        5     69.00    1,000  Free      0       Everyone      Education  
2750       34     12.00    5,000  Free      0       Everyone         Comics  
3030       24      3.20    5,000  Free      0       Everyone          Tools  
3115      103      3.80    5,000  Free      0       Everyone      Lifestyle  
4058       10      3.00   10,000  Free      0       Everyone  Entertainment  

[271 rows x 10 columns]

### What problem might you encounter if you rely exclusively on ratings alone to determine the quality of an app? ### 
271 apps with rating of 5, some with only 1 review, many with few - small sample size

# Find 5 Largest Apps in terms of Size (MBs)

**Challenge**: What's the size in megabytes (MB) of the largest Android apps in the Google Play Store. Based on the data, do you think there could be limit in place or can developers make apps as large as they please? 

In [54]:
df_apps_clean[df_apps_clean["Size_MBs"] == df_apps_clean["Size_MBs"].max()]

App            Category  \
1795                               Navi Radiography Pro             MEDICAL   
3144                                         Vi Trainer  HEALTH_AND_FITNESS   
4176   Car Crash III Beam DH Real Damage Simulator 2018                GAME   
7926                                          Post Bank             FINANCE   
7927                        The Walking Dead: Our World                GAME   
7928                      Stickman Legends: Shadow Wars                GAME   
8718                  Mini Golf King - Multiplayer Game                GAME   
8719                         Draft Simulator for FUT 18              SPORTS   
9942                     Talking Babsy Baby: Baby Games           LIFESTYLE   
9943                              Miami crime simulator                GAME   
9944                       Gangster Town: Vice District              FAMILY   
9945                                    Ultimate Tennis              SPORTS   
10295                                   SimCity BuildIt              FAMILY   
10687                            Hungry Shark Evolution                GAME   

       Rating  Reviews  Size_MBs     Installs  Type   Price Content_Rating  \
1795     4.70       11    100.00          500  Paid  $15.99       Everyone   
3144     3.60      124    100.00        5,000  Free       0       Everyone   
4176     3.60      151    100.00       10,000  Free       0       Everyone   
7926     4.50    60449    100.00    1,000,000  Free       0       Everyone   
7927     4.00    22435    100.00    1,000,000  Free       0           Teen   
7928     4.40    38419    100.00    1,000,000  Paid   $0.99   Everyone 10+   
8718     4.50   531458    100.00    5,000,000  Free       0       Everyone   
8719     4.60   162933    100.00    5,000,000  Free       0       Everyone   
9942     4.00   140995    100.00   10,000,000  Free       0       Everyone   
9943     4.00   254518    100.00   10,000,000  Free       0     Mature 17+   
9944     4.30    65146    100.00   10,000,000  Free       0     Mature 17+   
9945     4.30   183004    100.00   10,000,000  Free       0       Everyone   
10295    4.50  4218587    100.00   50,000,000  Free       0   Everyone 10+   
10687    4.50  6074334    100.00  100,000,000  Free       0           Teen   

                       Genres  
1795                  Medical  
3144         Health & Fitness  
4176                   Racing  
7926                  Finance  
7927                   Action  
7928                   Action  
8718                   Sports  
8719                   Sports  
9942   Lifestyle;Pretend Play  
9943                   Action  
9944               Simulation  
9945                   Sports  
10295              Simulation  
10687                  Arcade

### Do you think there could be a limit in place or can developers make apps as large as they please? ###
Definite limit of 100 Mb in place 

In [55]:
df_apps_clean.loc[df_apps_clean["Size_MBs"].idxmax()] # idxmax only returns first index at max value


App               Navi Radiography Pro
Category                       MEDICAL
Rating                            4.70
Reviews                             11
Size_MBs                        100.00
Installs                           500
Type                              Paid
Price                           $15.99
Content_Rating                Everyone
Genres                         Medical
Name: 1795, dtype: object

# Find the 5 App with Most Reviews

**Challenge**: Which apps have the highest number of reviews? Are there any paid apps among the top 50?

In [56]:
df_apps_clean["Reviews"]

21              4
28              2
47              2
82              7
99              1
           ...   
10824     2731171
10828    25655305
10829      906384
10831      877635
10835    27722264
Name: Reviews, Length: 8197, dtype: int64

In [57]:
display(df_apps_clean["Reviews"].nlargest(5)) # gets 5 most reviews and their index but not the rest of the info 
reviews = df_apps_clean["Reviews"].nlargest(5)
reviews.index[1] # can obtain the index from the nlargest

# better to just use sort values by reviews
df_apps_clean.sort_values(by=["Reviews"], ascending=False)


10805    78158306
10785    69119316
10806    66577313
10784    56642847
10650    44891723
Name: Reviews, dtype: int64

App            Category  \
10805                                        Facebook              SOCIAL   
10785                              WhatsApp Messenger       COMMUNICATION   
10806                                       Instagram              SOCIAL   
10784        Messenger – Text and Video Chat for Free       COMMUNICATION   
10650                                  Clash of Clans                GAME   
...                                               ...                 ...   
453                               Wowkwis aq Ka'qaquj              FAMILY   
462                                            CB Fit  HEALTH_AND_FITNESS   
1416   Ek Kahani Aisi Bhi Season 3 - The Horror Story              FAMILY   
485                             Familyfirst Messenger             MEDICAL   
1100                                     Chenoweth AH             MEDICAL   

       Rating   Reviews  Size_MBs       Installs  Type Price Content_Rating  \
10805    4.10  78158306      5.30  1,000,000,000  Free     0           Teen   
10785    4.40  69119316      3.50  1,000,000,000  Free     0       Everyone   
10806    4.50  66577313      5.30  1,000,000,000  Free     0           Teen   
10784    4.00  56642847      3.50  1,000,000,000  Free     0       Everyone   
10650    4.60  44891723     98.00    100,000,000  Free     0   Everyone 10+   
...       ...       ...       ...            ...   ...   ...            ...   
453      5.00         1     49.00             10  Free     0       Everyone   
462      5.00         1      7.80             10  Free     0       Everyone   
1416     3.00         1      5.80            100  Free     0           Teen   
485      4.00         1      3.30             10  Free     0       Everyone   
1100     5.00         1     27.00            100  Free     0       Everyone   

                    Genres  
10805               Social  
10785        Communication  
10806               Social  
10784        Communication  
10650             Strategy  
...                    ...  
453    Education;Education  
462       Health & Fitness  
1416         Entertainment  
485                Medical  
1100               Medical  

[8197 rows x 10 columns]

In [58]:
top_50 = df_apps_clean.sort_values(by=["Reviews"], ascending=False)[:50]
top_50["Price"] = pd.to_numeric(top_50["Price"]) # convert to numeric
top_50.dtypes 
# 2 different columns to check
top_50[top_50["Price"]!=0]
top_50[top_50["Type"] != "Free"]

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size_MBs, Installs, Type, Price, Content_Rating, Genres]
Index: []

# Plotly Pie and Donut Charts - Visualise Categorical Data: Content Ratings

In [59]:
ratings = df_apps_clean.Content_Rating.value_counts()
ratings

Everyone           6619
Teen                912
Mature 17+          357
Everyone 10+        305
Adults only 18+       3
Unrated               1
Name: Content_Rating, dtype: int64

In [63]:
# np.array(ratings)[:, None]
# ratings


Index(['Everyone', 'Teen', 'Mature 17+', 'Everyone 10+', 'Adults only 18+',
       'Unrated'],
      dtype='object')

In [61]:
# np.array(ratings)[:, None]
# ratings
# ratings_values = ratings.values.to_numpy()
fig = px.pie(labels=ratings.index.values, values= ratings.values) # index.values not index, turns it index into array  - else futurewarning
fig.show()

In [66]:
fig = px.pie(labels=ratings.index.values, values=ratings.values, title="Content Rating", names=ratings.index) # names gives category names
fig.update_traces(textposition='outside', textinfo= "percent+label")
fig.show()

# Numeric Type Conversion: Examine the Number of Installs

**Challenge**: How many apps had over 1 billion (that's right - BILLION) installations? How many apps just had a single install? 

Check the datatype of the Installs column.

Count the number of apps at each level of installations. 

Convert the number of installations (the Installs column) to a numeric data type. Hint: this is a 2-step process. You'll have make sure you remove non-numeric characters first. 

# Find the Most Expensive Apps, Filter out the Junk, and Calculate a (ballpark) Sales Revenue Estimate

Let's examine the Price column more closely.

**Challenge**: Convert the price column to numeric data. Then investigate the top 20 most expensive apps in the dataset.

Remove all apps that cost more than $250 from the `df_apps_clean` DataFrame.

Add a column called 'Revenue_Estimate' to the DataFrame. This column should hold the price of the app times the number of installs. What are the top 10 highest grossing paid apps according to this estimate? Out of the top 10 highest grossing paid apps, how many are games?


### The most expensive apps sub $250

### Highest Grossing Paid Apps (ballpark estimate)

# Plotly Bar Charts & Scatter Plots: Analysing App Categories

### Vertical Bar Chart - Highest Competition (Number of Apps)

### Horizontal Bar Chart - Most Popular Categories (Highest Downloads)

### Category Concentration - Downloads vs. Competition

**Challenge**: 
* First, create a DataFrame that has the number of apps in one column and the number of installs in another:

<img src=https://imgur.com/uQRSlXi.png width="350">

* Then use the [plotly express examples from the documentation](https://plotly.com/python/line-and-scatter/) alongside the [.scatter() API reference](https://plotly.com/python-api-reference/generated/plotly.express.scatter.html)to create scatter plot that looks like this. 

<img src=https://imgur.com/cHsqh6a.png>

*Hint*: Use the size, hover_name and color parameters in .scatter(). To scale the yaxis, call .update_layout() and specify that the yaxis should be on a log-scale like so: yaxis=dict(type='log') 

# Extracting Nested Data from a Column

**Challenge**: How many different types of genres are there? Can an app belong to more than one genre? Check what happens when you use .value_counts() on a column with nested values? See if you can work around this problem by using the .split() function and the DataFrame's [.stack() method](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.stack.html). 


# Colour Scales in Plotly Charts - Competition in Genres

**Challenge**: Can you create this chart with the Series containing the genre data? 

<img src=https://imgur.com/DbcoQli.png width=400>

Try experimenting with the built in colour scales in Plotly. You can find a full list [here](https://plotly.com/python/builtin-colorscales/). 

* Find a way to set the colour scale using the color_continuous_scale parameter. 
* Find a way to make the color axis disappear by using coloraxis_showscale. 

# Grouped Bar Charts: Free vs. Paid Apps per Category

**Challenge**: Use the plotly express bar [chart examples](https://plotly.com/python/bar-charts/#bar-chart-with-sorted-or-ordered-categories) and the [.bar() API reference](https://plotly.com/python-api-reference/generated/plotly.express.bar.html#plotly.express.bar) to create this bar chart: 

<img src=https://imgur.com/LE0XCxA.png>

You'll want to use the `df_free_vs_paid` DataFrame that you created above that has the total number of free and paid apps per category. 

See if you can figure out how to get the look above by changing the `categoryorder` to 'total descending' as outlined in the documentation here [here](https://plotly.com/python/categorical-axes/#automatically-sorting-categories-by-name-or-total-value). 

# Plotly Box Plots: Lost Downloads for Paid Apps

**Challenge**: Create a box plot that shows the number of Installs for free versus paid apps. How does the median number of installations compare? Is the difference large or small?

Use the [Box Plots Guide](https://plotly.com/python/box-plots/) and the [.box API reference](https://plotly.com/python-api-reference/generated/plotly.express.box.html) to create the following chart. 

<img src=https://imgur.com/uVsECT3.png>


# Plotly Box Plots: Revenue by App Category

**Challenge**: See if you can generate the chart below: 

<img src=https://imgur.com/v4CiNqX.png>

Looking at the hover text, how much does the median app earn in the Tools category? If developing an Android app costs $30,000 or thereabouts, does the average photography app recoup its development costs?

Hint: I've used 'min ascending' to sort the categories. 

# How Much Can You Charge? Examine Paid App Pricing Strategies by Category

**Challenge**: What is the median price price for a paid app? Then compare pricing by category by creating another box plot. But this time examine the prices (instead of the revenue estimates) of the paid apps. I recommend using `{categoryorder':'max descending'}` to sort the categories.